tests/test_mcp.py

```diff
@@ -42,6 +46,71 @@ def agent(openai_api_key: str):
     return Agent(model, mcp_servers=[server])
 
 
+def run_sse_server() -> None:
+    """Run the Streamable HTTP MCP server."""
+    test_mcp_server.run(transport='sse')
+
+
+def run_streamable_http_server() -> None:
+    """Run the SSE MCP Server"""
+    test_mcp_server.run(transport='streamable-http')
+
+
+@pytest.fixture
+def mcp_server(request: pytest.FixtureRequest):
+    proc = multiprocessing.Process(target=request.param, daemon=True)
+    print('Staring streamable http server process on port port')
+    proc.start()
+
+    max_attempts = 20
+    attempt = 0
+
+    while attempt < max_attempts:
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.connect(('localhost', 8000))
```

Comment:
We should use a randomly assigned port here, we can't be sure 8000 will be available

Reply:
It looks like FastMCP has its own host and port settings, and the `run` method runs the entire server. Do we need to run it in a separate process at all? Could we use a thread? It looks like `FastMCP.run` calls `anyio.run(self.run_sse_async)`. We could likely do something ourselves to run the `run_sse_async` async method in a thread.